nexus/src/app/oximeter.rs

```diff
@@ -212,14 +216,100 @@ impl super::Nexus {
             .map_err(|e| Error::internal_error(&e.to_string()))?
             .timeseries_schema_list(&pag_params.page, limit)
             .await
-            .map_err(|e| match e {
-                oximeter_db::Error::DatabaseUnavailable(_) => {
-                    Error::ServiceUnavailable {
-                        internal_message: e.to_string(),
-                    }
+            .map_err(map_oximeter_err)
+    }
+
+    /// Returns a results from the timeseries DB based on the provided query
+    /// parameters.
+    ///
+    /// * `timeseries_name`: The "target:metric" name identifying the metric to
+    /// be queried.
+    /// * `criteria`: Any additional parameters to help narrow down the query
+    /// selection further.
+    /// * `query_params`: Pagination parameter, identifying which page of
+    /// results to return.
+    /// * `limit`: The maximum number of results to return in a paginated
+    /// request.
+    pub async fn select_timeseries(
+        &self,
+        timeseries_name: &str,
+        criteria: &[&str],
+        query_params: ResourceMetricsPagination,
+        limit: NonZeroU32,
+    ) -> Result<dropshot::ResultsPage<Measurement>, Error> {
+        #[inline]
+        fn no_results() -> dropshot::ResultsPage<Measurement> {
+            dropshot::ResultsPage { next_page: None, items: Vec::new() }
+        }
+
+        let (start_time, end_time, query) = match query_params.page {
+            dropshot::WhichPage::First(query) => (
+                Timestamp::Inclusive(query.start_time),
+                Timestamp::Exclusive(query.end_time),
+                query,
+            ),
+            dropshot::WhichPage::Next(query) => (
+                Timestamp::Exclusive(query.start_time),
```

Comment:
This seems like it should be inclusive. The first page query accepts a time range `[t0, t1)`. Then later pages have `(t1, t2)`, `(t2, t3)`, etc. It's unlikely, but that seems to mean we'd miss samples lying right on those boundaries. Should all pages just return `[t0, t1)`?

Reply:
Looking at how you generate the pagination parameters, I think I see why this is exclusive of both ends for later pages. We're taking the last timestamp we _actually_ observed from the database as the key, and so we want to make sure to exclude that on later pages. Assuming that's right, then this seems OK, though probably worth a quick comment here.

Reply:
Yeah, the tests caught this one for me. I'll add a comment explaining things.